After v1.14.33, OpenCode Desktop on Windows no longer respects proxy settings for GitHub Copilot authentication.

### Description

After v1.14.33, OpenCode Desktop on Windows no longer respects proxy settings for GitHub Copilot authentication.

Environment:
- OS: Windows
- OpenCode Desktop:
  - v1.14.33: works
  - v1.14.34+ broken
- Provider: GitHub Copilot
- Proxy is configured via HTTP_PROXY / HTTPS_PROXY.
- The same proxy configuration works in v1.14.33.

Observed behavior:
Starting from versions after v1.14.33, GitHub Copilot provider authentication fails with:

Authorization failed: TypeError: fetch failed

Logs show that OpenCode attempts a direct connection instead of using the configured proxy:

fetch failed
Caused by: Connect Timeout Error
attempted address: github.com:443
timeout: 10000ms

Expected behavior:
OpenCode Desktop should use the configured proxy for GitHub/Copilot authentication requests, as it did in v1.14.33.

Workaround:
Downgrading to v1.14.33 fixes the issue.

### Plugins

No plugins / default setup

### OpenCode version

* v1.14.33: works
* v1.14.34: broken
* v1.14.35: broken
* v1.14.37: broken

### Steps to reproduce

1. On Windows, configure proxy environment variables: 
```
HTTP_PROXY=http://127.0.0.1:<proxy-port>
HTTPS_PROXY=http://127.0.0.1:<proxy-port>
NO_PROXY=localhost,127.0.0.1
```
2. Install OpenCode Desktop v1.14.33.
3. Open OpenCode Desktop.
4. Configure/sign in to GitHub Copilot provider.
5. Confirm that GitHub Copilot authentication works.
6. Upgrade OpenCode Desktop to v1.14.34 or later
7. Try to authenticate GitHub Copilot provider again.
8. Authentication fails with:
`Authorization failed: TypeError: fetch failed`
9. Check OpenCode logs.
10. The logs show that OpenCode tries to connect directly to github.com:443 and times out instead of using the configured proxy.


### Screenshot and/or share link

<img width="814" height="367" alt="Image" src="https://github.com/user-attachments/assets/e953d8a1-1e0b-45e2-9711-3e00fd90f7aa" />

```
ERROR 2026-05-05T08:58:52 +889ms service=file.watcher error=Cannot find module '@parcel/watcher-win32-x64'
Require stack:
- C:\Users\<user>\AppData\Local\Programs\OpenCode\resources\app.asar\out\main\chunks\node-DJvLtQ6Z.js failed to load watcher binding
ERROR 2026-05-05T08:59:02 +9981ms service=default Failed to fetch models.dev [object Object]
ERROR 2026-05-05T08:59:28 +25879ms service=server error=fetch failed Caused by: Connect Timeout Error (attempted address: github.com:443, timeout: 10000ms) failed
```

### Operating System

Windows 11

### Terminal

OpenCode Desktop on Windows

--- a/packages/desktop/src/main/index.ts
+++ b/packages/desktop/src/main/index.ts
@@ -200,7 +200,10 @@ async function initialize() {
     }
 
     logger.log("spawning sidecar", { url })
-    const { listener, health } = await spawnLocalServer(hostname, port, password)
+    const { listener, health } = await spawnLocalServer(hostname, port, password, () => {
+      ensureLoopbackNoProxy()
+      useEnvProxy()
+    })
     server = listener
     serverReady.resolve({
       url,

--- a/packages/desktop/src/main/index.ts
+++ b/packages/desktop/src/main/index.ts
@@ -137,8 +137,8 @@ function useSystemCertificates() {
 
 function useEnvProxy() {
   try {
-    const setGlobalProxyFromEnv = (http as typeof http & { setGlobalProxyFromEnv: () => void }).setGlobalProxyFromEnv
-    setGlobalProxyFromEnv()
+    // Electron 41.2 runs Node 24.14.1; latest @types/node@24 is 24.12.2.
+    ;(http as any).setGlobalProxyFromEnv()
   } catch (error) {
     logger.warn("failed to load proxy environment", error)
   }

--- a/packages/desktop/src/main/index.ts
+++ b/packages/desktop/src/main/index.ts
@@ -1,6 +1,7 @@
 import { randomUUID } from "node:crypto"
 import { EventEmitter } from "node:events"
 import { existsSync } from "node:fs"
+import * as http from "node:http"
 import { createServer } from "node:net"
 import { homedir } from "node:os"
 import { join } from "node:path"
@@ -77,6 +78,7 @@ setupApp()
 
 function setupApp() {
   ensureLoopbackNoProxy()
+  useEnvProxy()
   app.commandLine.appendSwitch("proxy-bypass-list", "<-loopback>")
   if (!app.isPackaged) app.commandLine.appendSwitch("remote-debugging-port", "9222")
 
@@ -130,6 +132,15 @@ function useSystemCertificates() {
     setDefaultCACertificates([...new Set([...getCACertificates("default"), ...getCACertificates("system")])])
   } catch (error) {
     logger.warn("failed to load system certificates", error)
+  }
+}
+
+function useEnvProxy() {
+  try {
+    const setGlobalProxyFromEnv = (http as typeof http & { setGlobalProxyFromEnv: () => void }).setGlobalProxyFromEnv
+    setGlobalProxyFromEnv()
+  } catch (error) {
+    logger.warn("failed to load proxy environment", error)
   }
 }
 

--- a/packages/desktop/src/main/server.ts
+++ b/packages/desktop/src/main/server.ts
@@ -30,8 +30,9 @@ export function setWslConfig(config: WslConfig) {
   getStore().set(WSL_ENABLED_KEY, config.enabled)
 }
 
-export async function spawnLocalServer(hostname: string, port: number, password: string) {
+export async function spawnLocalServer(hostname: string, port: number, password: string, configureEnv?: () => void) {
   prepareServerEnv(password)
+  configureEnv?.()
   const { Log, Server } = await import("virtual:opencode-server")
   await Log.init({ level: "WARN" })
   const listener = await Server.listen({

